### ISSUE TITLE
Problem with Checking whether a User is Logged in

### ISSUE THREAD

**Issue by seonghoon247** [swsnu/swpp17-team6#17]
The userinfo method in the user/views.py will send the user info if the user exists,  
but will send status code 401 if the user does not exist.

```
def userinfo(request):
	if request.method == 'GET':
		# get userinfo - assume logged in
		if request.user is not None:
			userinfo = UserInfo.objects.get(id = request.user.id)
			result = {}
			result['email'] = request.user.username
			result['username'] = userinfo.nickname
			result['locations'] = getLocationStr(userinfo)
			result['services'] = getServiceStr(userinfo)
			return JsonResponse (result)
		else:
			return HttpResponse(status = 401)
```

However, the current frontend implementation for checking user's login status (shown as below) does not work as expected.
```
    ngOnInit(): void{
        this.userService.getUser().then(user => {
            if(user == null)    this.router.navigate(['/signin']);
        })
    }
```

I believe this is because if the user does not exist, the backend will **NOT send NULL**,  
but instead send a response code.

**Comment by HS-YN** [comment]
Changing the `ngOnInit()` part from 'checking the user is null' to 'checking the status is 401' would resolve the issue. I will try handling this issue as soon as possible.

**Comment by seonghoon247** [comment]
Since the current getUser() method returns a User (in promise), it cannot receive the status code as its return value. Is there a way for a function to dynamically decide the return type?
```
    getUser(): Promise<User> {
        return this.http.get(this.userInfoUrl).toPromise().then(response => 
            response.json() as User);
    }
```

**Comment by HS-YN** [comment]
Since javascript is a weakly-typed language, it does dynamically decide the type in a sense (i.e. type `any`. But in this case, I think
1. Chaining promise method (if return value is number, do one thing and do another thing otherwise)
2. Defining another api & method for session checking
one of these solutions would work.

**Comment by seonghoon247** [comment]
I think we can solve this issue by creating a separate user service method in the frontend, which will receive 200 or 401 depending on success and failure

**Comment by seonghoon247** [comment]
Resolved along with #28
